app/src/main/java/com/android/shelfLife/ui/newoverview/NewFoodDisplayComponents.kt

```diff
@@ -0,0 +1,179 @@
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.ElevatedCard
+import androidx.compose.material3.LinearProgressIndicator
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color.Companion.LightGray
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import coil3.compose.AsyncImage
+import com.android.shelfLife.model.foodFacts.FoodUnit
+import com.android.shelfLife.model.newFoodItem.FoodItem
+import com.android.shelfLife.model.newFoodItem.FoodStatus
+import com.android.shelfLife.model.newOverview.OverviewScreenViewModel
+import com.android.shelfLife.ui.utils.getExpiryMessageBasedOnDays
+import com.android.shelfLife.ui.utils.getProgressBarState
+import com.google.firebase.Timestamp
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+/**
+ * Composable function to display the list of food items
+ *
+ * @param foodItems The list of food items to display
+ */
+@Composable
+fun ListFoodItems(
+    foodItems: List<FoodItem>,
+    overviewScreenViewModel: OverviewScreenViewModel,
+    onFoodItemClick: (FoodItem) -> Unit,
+    onFoodItemLongHold: (FoodItem) -> Unit
+) {
+  if (foodItems.isEmpty()) {
+    Box(
+        modifier = Modifier.fillMaxSize().testTag("NoFoodItems"),
+        contentAlignment = Alignment.Center) {
+          Text(text = "No food available")
+        }
+  } else {
+    // Display the full list
+    LazyColumn(modifier = Modifier.fillMaxSize().testTag("foodItemList")) {
+      items(foodItems) { item ->
+        // Call a composable that renders each individual to-do item
+        FoodItemCard(
+            foodItem = item,
+            overviewScreenViewModel = overviewScreenViewModel,
+            onClick = { onFoodItemClick(item) },
+            onLongPress = { onFoodItemLongHold(item) })
+      }
+    }
+  }
+}
+
+@OptIn(ExperimentalFoundationApi::class)
+@Composable
+fun FoodItemCard(
```

Comment:
This class seems to have a lot of logic that could go in the view model. I am not sure but I think its something we could talk about

Reply:
Yes, I did not write it and I agree